mmdet3d/datasets/semantickitti_dataset.py

```diff
@@ -108,3 +141,281 @@ def get_ann_info(self, index):
+            gt_sem = gt & 0xFFFF
+            gt_sem = np.vectorize(class_remap.__getitem__)(gt_sem)
+            gt_annos.append(dict(gt_sem=gt_sem, gt_inst=gt))
+
+        return gt_annos
+
+    def evaluate(self,
+                 results,
+                 metric=None,
+                 logger=None,
+                 pklfile_prefix=None,
+                 submission_prefix=None,
+                 show=False,
+                 out_dir=None,
+                 pipeline=None):
+        """Evaluation in KITTI protocol.
+
+        Args:
+            results (list[dict]): Testing results of the dataset.
+            metric (str | list[str], optional): Metrics to be evaluated.
+                Default: None.
+            logger (logging.Logger | str, optional): Logger used for printing
+                related information during evaluation. Default: None.
+            pklfile_prefix (str, optional): The prefix of pkl files, including
+                the file path and the prefix of filename, e.g., "a/b/prefix".
+                If not specified, a temp file will be created. Default: None.
+            submission_prefix (str, optional): The prefix of submission data.
+                If not specified, the submission data will not be generated.
+                Default: None.
+            show (bool, optional): Whether to visualize.
+                Default: False.
+            out_dir (str, optional): Path to save the visualization results.
+                Default: None.
+            pipeline (list[dict], optional): raw data loading for showing.
+                Default: None.
+
+        Returns:
+            dict[str, float]: Results of each evaluation metric.
+        """
+        result_files, tmp_dir = self.format_results(results, pklfile_prefix)
+
+        # get the labels
+        gt_files = [info['pts_semantic_mask_path'] for info in self.data_infos]
+        gt_annos = self.parse_label(gt_files)
+
+        if isinstance(result_files, dict):
+            ap_dict_list = []
+            ap_dict = dict()
+            for name, result_files_ in result_files.items():
+
+                ap_result_str, ap_dict_ = self.semantic_kitti_eval(
+                    gt_annos, result_files_, self.CLASSES)
+                for ap_type, ap in ap_dict_.items():
+                    ap_dict[f'{name}/{ap_type}'] = float('{:.4f}'.format(ap))
+                ap_dict_list.append(ap_dict)
+                print_log(
+                    f'Results of {name}:\n' + ap_result_str, logger=logger)
+
+        else:
+            ap_result_str, ap_dict_list = self.semantic_kitti_eval(
+                gt_annos, result_files)
+            print_log('\n' + ap_result_str, logger=logger)
+
+        if tmp_dir is not None:
+            tmp_dir.cleanup()
+        if show or out_dir:
+            self.show(results, out_dir, show=show, pipeline=pipeline)
+        return ap_dict_list
+
+    def semantic_kitti_eval(self, gt_annos, result_files):
```

Comment:
These specific evaluation codes can be moved to mmdet3d/core/evaluation/semantickitti_eval.py like other datasets.

Reply:
I have migrated the evaluation codes to the above path, but by a way of `class SemanticKITTIEval(object)`. Is that ok?